kafka-client/src/main/java/dev/responsive/kafka/internal/stores/LocalRemoteKvIterator.java

```diff
@@ -121,13 +133,21 @@ private KeyValue<K, byte[]> advance() {
       // return that
       return remote.next();
     } else if (remoteKey.compareTo(cachedKey) == 0) {
-      // if they're the same, then there are two options:
-      // (1) the value is a tombstone, in which case we
-      // discard both keys and move on or (2) the cached
-      // value is more recent, so we return that and advance
-      // both - either way, the value from remote.next()
-      // should not be returned
-      remote.next();
+
+      if (retainDuplicates) {
+        // if keys are the same and duplicates are allowed, then we arbitrarily
+        // pick one of them to return (in this case the remote) but it doesn't
+        // really matter which
+        return remote.next();
```

Comment:
Took me a while to realize that we're supposed to return _both_ it's just the order that doesn't matter. Can we add that as a comment?

Reply:
ah, yes that's fair, will do